iree/compiler/Dialect/VM/Transforms/Passes.cpp

```diff
@@ -30,6 +30,10 @@ void buildVMTransformPassPipeline(OpPassManager &passManager,
                                   TargetOptions targetOptions) {
   passManager.addPass(createCanonicalizerPass());
   passManager.addPass(createConversionPass(targetOptions));
+  // The createGlobalInitializationPass operates on vm.module ops, so requires
+  // that the dialect already be registered to be added to the pass pipeline.
+  // TODO(#2958): There should be a better way to do this.
+  passManager.getContext()->loadDialect<VM::VMDialect>();
```

Comment:
anything dealing with the VM/ files needs the dialect registered - why is this needed here specifically?

Reply:
This loads it explicitly, which is required before this pass can be instantiated. I also think it's bad that this is required and I think Mehdi was going to investigate why it was actually necessary.

Reply:
It is necessary because the pass isn't an operation pass, a ModulePass, or a Function pass.
I don't see an easy way to solve this right now: we can't retrieve a TypeID for the dialect from an operation right now.

Reply:
The dialect is already registered (and also has to be for us to load it here, right?)

Reply:
No the dialect does not need to be registered for you to load it here.

Reply:
My point is that without the dialect being present one cannot even load the module (how else are custom parsers found?), and we load the module prior to creating a pass manager and running it, so I'm a bit lost as to why this line of code is required. It seems random. If it is required here, I feel like that's a design issue in the registration system that should be worked out - needing to do this is extremely intuitive.

Reply:
Looks like https://reviews.llvm.org/D86915 is the fix, thanks!